common/http/platform/beast/http.cpp

```diff
@@ -188,18 +188,17 @@ error::Error Client::AsyncCall(
 	body_handler_ = body_handler;
 	status_ = TransactionStatus::None;
 
-	// See comment in header.
-	client_active_.reset(this, [](Client *) {});
+	*cancelled_ = false;
 
-	weak_ptr<Client> weak_client(client_active_);
+	auto &cancelled = cancelled_;
```

Comment:
After a reference has been assigned, it behaves as if it were the original object, so using a reference here does not automatically trigger the use of a reference in the lambda. For that we would have to use the `[&cancelled]` expression. So in other words, there is another copy being made when we capture the variable in the lambda, and using a reference here is just a micro-optimization to avoid making two copies.

Reply:
Aha! Thank you.

Reply:
C++ at its finest :laughing: